src/Common/MoistThermodynamics/MoistThermodynamics.jl

```diff
@@ -737,6 +738,52 @@ PhasePartition(ts::PhaseDry{FT}) where {FT<:Real} = PhasePartition(FT(0), FT(0),
 PhasePartition(ts::PhaseEquil) = PhasePartition_equil(air_temperature(ts), air_density(ts), ts.q_tot)
 PhasePartition(ts::PhaseNonEquil) = ts.q
 
+function roots_∂e_int_∂T(T::FT, e_int::FT, ρ::FT, q_tot::FT) where {FT<:Real}
+  cvm = cv_m(PhasePartition(q_tot))
+  q_vap_sat = q_vap_saturation(T, ρ)
```

Comment:
We should also avoid computing `q_vap_sat` twice, once here and once in `PhasePartition`. Either make it an optional input argument to `PhasePartition`, so it does not need to be computed there, or make it an optional output argument, so it does not need to be computed here.

Reply:
`PhasePartition` is a constructor and doesn't call `q_vap_saturation`. Do you mean in `PhasePartition_equil`? We can do that, I'll add an issue to fix that (it'll involve changing a handful of function calls, but it's a good idea since it may result in saved flops in other function calls).

Reply:
Yes, the equilibrium version.

Reply:
(The equilibrium version is what you need here.)